src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -217,15 +216,12 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
                         return Ok((res, Some(path_str.to_owned())));
                     }
                     Res::Def(DefKind::Mod, _) => {
-                        // This resolved to a module, but if we were passed `type@`,
-                        // we want primitive types to take precedence instead.
-                        if disambiguator == Some(Disambiguator::Namespace(Namespace::TypeNS)) {
-                            if let Some(prim) = is_primitive(path_str, ns) {
-                                if extra_fragment.is_some() {
-                                    return Err(ErrorKind::AnchorFailure(AnchorFailure::Primitive));
-                                }
-                                return Ok((prim, Some(path_str.to_owned())));
+                        // This resolved to a module, but we want primitive types to take precedence instead.
```

Comment:
We should avoid this branch if it's explicitly not specified as a type, yes?

Reply:
Oh you're right - this should happen if there's no disambiguator or if the disambiguator is `type@`, but if there's a different ambiguator we should skip it.

Reply:
Should be fixed now.